### PR TITLE
Add skip_old_times option to Predictor

### LINKED ISSUE #6556: Option to skip predictor for a list of old times
### Description of the enhancement or error report
When started using the feature developed in #6506, I realized that what I really want is to specify a list of old times for which I want to skip the predictor.  We often use piecewise linear functions to define a loading history, and they have kinks in them at the points that define the function. It's the step following that kink where we need to skip the predictor, not the step that ends on that kink.

### Rationale for the enhancement or information for reproducing the error
This will improve the utility of the predictor for non-monotonic loading scenarios.

### Identified impact
The Predictor base class

### PR DESCRIPTION
**Pull request by bwspenc**
closes #6556

### CODE REVIEW

**Comment by andrsd** [comment]
Do you assume that people would want to skip current and old time steps at the same time? If not, then you could use 'skip_times' vector with a flag that would apply it to the old time steps... Just a thought...

**Comment by bwspenc** [comment]
@andrsd, sure, I'm open to suggestions, and that's a reasonable one. I'm not totally crazy about the naming of the ```skip_times``` and ```skip_old_times``` parameters.

**Comment by bwspenc** [comment]
@andrsd After thinking this over a little more, I think it would be reasonable to allow users to specify a combination of old times and new times for which they would want to skip the predictor. How about these two parameters: ```skip_when_previous_time =``` and ```skip_when_current_time = ```?

**Comment by andrsd** [comment]
`skip_times` and `skip_times_old` like `coupledValue`, `coupledValueOld`?
I do not really have strong opinion on this :smiley:

**Comment by bwspenc** [comment]
OK, I like ```skip_times``` and ```skip_times_old```.  I just updated it so that it is implemented that way. If nobody objects to the names, I think this is good to go.

File: framework/src/predictors/Predictor.C [hunk 1]
```diff
@@ -67,13 +69,16 @@ Predictor::shouldApply()
   bool should_apply = true;
 
   const Real & current_time =  _fe_problem.time();
-  for (unsigned int i=0; i<_skip_times.size(); ++i)
+  const Real & old_time =  _fe_problem.timeOld();
```
Comment (andrsd) [inline framework/src/predictors/Predictor.C]: Reduce 2 spaces into 1 ;-)

Comment (bwspenc) [inline framework/src/predictors/Predictor.C]: Just overachieving on my spacing around operators!  It's fixed now.

**Pull request merged**